### PR TITLE
fix: timeout docs + types

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
@Ethan-Arrowood

**Comment by ronag** [comment]
@Ethan-Arrowood 

Can you help me with the errors?

```
  test/types/client.test-d.ts:52:66
  ✖  52:66  Argument of type { path: string; } is not assignable to parameter of type UpgradeOptions.
  Property method is missing in type { path: string; } but required in type UpgradeOptions.  
  ✖  53:39  Argument of type { path: string; } is not assignable to parameter of type UpgradeOptions.
  Property method is missing in type { path: string; } but required in type UpgradeOptions.  
  ✖  59:66  Argument of type { path: string; } is not assignable to parameter of type ConnectOptions.
  Property method is missing in type { path: string; } but required in type ConnectOptions.  
  ✖  60:39  Argument of type { path: string; } is not assignable to parameter of type ConnectOptions.
  Property method is missing in type { path: string; } but required in type ConnectOptions.  

  test/types/pool.test-d.ts:52:64
  ✖  52:64  Argument of type { path: string; } is not assignable to parameter of type UpgradeOptions.
  Property method is missing in type { path: string; } but required in type UpgradeOptions.  
  ✖  53:37  Argument of type { path: string; } is not assignable to parameter of type UpgradeOptions.
  Property method is missing in type { path: string; } but required in type UpgradeOptions.  
  ✖  59:64  Argument of type { path: string; } is not assignable to parameter of type ConnectOptions.
  Property method is missing in type { path: string; } but required in type ConnectOptions.  
  ✖  60:37  Argument of type { path: string; } is not assignable to parameter of type ConnectOptions.
  Property method is missing in type { path: string; } but required in type ConnectOptions.  
```

**Comment by ronag** [comment]
review with "Hide whitespace changes" enabled.

File: types/client.d.ts [hunk 1]
```diff
@@ -4,211 +4,171 @@ import { Duplex, Readable, Writable } from 'stream'
...
 import { IncomingHttpHeaders } from 'http'
 
-/**
- * Issue https://github.com/nodejs/undici/issues/480
- * 
- * AbortController is not a dependency of Undici. It is a new feature in Node 15, and requires a shim to use in prior Node versions.
- * For now, this type will be assigned as `unknown`. When @types/node ships a definition for it we will modify this definition.
- * Until then, TS devs should also rely on the shim `abort-controller` if they want the correct type definitions.
- */
-type AbortController = { abort: unknown };
```
Comment (Ethan-Arrowood) [inline types/client.d.ts:14]: Why was this removed?

File: types/client.d.ts [hunk 2]
```diff
@@ -4,211 +4,171 @@ import { Duplex, Readable, Writable } from 'stream'
...
+    headers: IncomingHttpHeaders;
+    opaque: unknown;
+    body: Readable;
+  }
+
+  export type PipelineHandler = (data: PipelineHandlerData) => Readable
+
+  export interface DispatchHandlers {
+    /** Invoked before request is dispatched on socket. May be invoked multiple times when a request is retried when the request at the head of the pipeline fails. */
+    onConnect?(abort: () => void): void;
```
Comment (Ethan-Arrowood) [inline types/client.d.ts:162]:
```suggestion
    onConnect?(abort: AbortController['abort']): void;
```

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]
What is this PR fixing exactly? I have a classic case of "it works on my machine" so I'm not sure whats going on

File: types/client.d.ts (hunk 2 above)
Comment (ronag) [reply types/client.d.ts:162]: It’s not an abortcontroller

Comment (Ethan-Arrowood) [reply types/client.d.ts:162]: Oh! This is a generic _abort_ not a AbortController.abort. Gotcha! Then can we still keep the comment but also remove the `type AbortController`?

**Comment by Ethan-Arrowood** [comment]
I also see now the doc changes are in master already (https://github.com/nodejs/undici/commit/4b6421f11e6c1bc4eff33c1561590cc247e73725)

So i'll review with that in mind.

**Comment by Ethan-Arrowood** [comment]
> Can you help me with the errors?

It looks like the `method` property is missing from the tests. I'll take a peak at whats on master now.

File: types/client.d.ts [hunk 3]
```diff
@@ -4,211 +4,171 @@ import { Duplex, Readable, Writable } from 'stream'
...
+    /** Default: `null` */
+    signal?: AbortSignal | EventEmitter | null;
+  }
+
+  export interface PipelineOptions extends RequestOptions {
+    /** `true` if the `handler` will return an object stream. Default: `false` */
+    objectMode?: boolean;
+  }
+
+  export interface UpgradeOptions extends DispatchOptions {
```
Comment (Ethan-Arrowood) [inline types/client.d.ts:111]: Per the test errors, previously `method` is optional for `upgrade`. But now that its extending from `DispatchOptions` it has become required. 

You can specify `method?: string` here to make it optional again

File: types/client.d.ts [hunk 4]
```diff
@@ -4,211 +4,171 @@ import { Duplex, Readable, Writable } from 'stream'
...
+    /** A string of comma separated protocols, in descending preference order. Default: `'Websocket'` */
+    protocol?: string;
+  }
+
+  export interface ConnectOptions extends DispatchOptions {
```
Comment (Ethan-Arrowood) [inline types/client.d.ts:116]: Similarly, `ConnectOptions` previously had no `method` property (but `DispatchOptions` does). 

If this was an error in the previous types then we just need to fix the tests and fix the documentation for the `connect` method.

If the is intended, then we could try assigning `method: never;`. Not sure how that well merge with `DispatchOptions` though. We might need to do something different like `Omit<'method', DispatchOptions>` instead.

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

File: types/client.d.ts (hunk 1 above)
Comment (ronag) [reply types/client.d.ts:14]: not required.

File: types/client.d.ts (hunk 2 above)
Comment (ronag) [reply types/client.d.ts:181]: what comment?
